src/common/device_helpers.cuh

```diff
@@ -429,7 +429,7 @@ struct XGBDefaultDeviceAllocatorImpl : XGBBaseDeviceAllocator<T> {
   }
 #if defined(XGBOOST_USE_RMM) && XGBOOST_USE_RMM == 1
   XGBDefaultDeviceAllocatorImpl()
-    : SuperT(rmm::mr::get_current_device_resource(), cudaStream_t{nullptr}) {}
+    : SuperT(cudaStream_t{nullptr}, rmm::mr::get_current_device_resource()) {}
```

Comment:
Recommend using `rmm::cuda_stream_default` here.

```suggestion
    : SuperT(rmm::cuda_stream_default, rmm::mr::get_current_device_resource()) {}
```

Reply:
Is there a semantic difference, or is this a cosmetic change?

Reply:
Ah, but this would not be backwards compatible.  To @pseudotensor 's point, I think the change to the parameter order of `rmm::mr::thrust_allocator_adaptor` was a breaking change. If you want backwards compatibility, you will need to detect the version of RMM and ifdef or something...

Reply:
@harrism Should XGBoost aim to support multiple versions of RMM? If so, is there a macro I can use to detect the version of RMM?

Reply:
> Is there a semantic difference, or is this a cosmetic change?

It's not cosmetic. In RMM and other RAPIDS C++ APIs we aim for type safety wherever possible. `cuda_stream_view` is a strongly typed wrapper for streams. Without it, you can end up with ambiguity, e.g. `SuperT(0, 0)` or `SuperT(nullptr, nullptr)` is perfectly valid and will compile. Even `SuperT(nullptr, cudaStreamDefault)` will probably compile. We added strongly typed wrappers to RMM to help improve RAPIDS APIs.

> @harrism Should XGBoost aim to support multiple versions of RMM? If so, is there a macro I can use to detect the version of RMM?

That's up to you. No, RMM doesn't currently have a version macro. This is something we could consider if you want to file an issue. Sorry about the breaking change.